### PR TITLE
DAT-4203 fix markers regex, add tests, intorduce loggin on error

### PR DESCRIPTION
**Pull request by idradm**
https://wikia-inc.atlassian.net/browse/DAT-4203

### CODE REVIEW

File: includes/wikia/parser/templatetypes/handlers/NavboxTemplate.class.php [hunk 1]
```diff
@@ -28,20 +28,30 @@ private static function process( $html ) {
...
+		);
+		if ( $replaced !== null ) {
 			// remove just the marker tags
-			$html = preg_replace( '/(<|&lt;)\\/?' . $marker . '(>|&gt;)/sU', '', $html );
+			$result = preg_replace( '/(<|&lt;)\\/?' . $marker . '(>|&gt;)/sU', '', $replaced );
+			if ( $result !== null ) {
+
+				return $result;
+			}
+			\Wikia\Logger\WikiaLogger::instance()->error( 'Navbox marker removal failed.' );
```
Comment (sebastianmarzjan) [inline includes/wikia/parser/templatetypes/handlers/NavboxTemplate.class.php]: Let's log the preg_last_error() too both here and in L54

File: includes/wikia/parser/templatetypes/handlers/NavboxTemplate.class.php [hunk 2]
```diff
@@ -28,20 +28,30 @@ private static function process( $html ) {
...
-				'<div data-type="navbox">$0</div>',
-				$html
-			);
+			$html = static::replaceMarker( $marker, $html );
+		}
+
+		return $html;
+	}
+
+	private static function replaceMarker( $marker, $html ) {
```
Comment (dianafa) [inline includes/wikia/parser/templatetypes/handlers/NavboxTemplate.class.php:37]: can we add an info that this is an experiment and should be cleaned up with ticket DAT-XYZ

Comment (idradm) [inline includes/wikia/parser/templatetypes/handlers/NavboxTemplate.class.php:37]: It's in the class description

File: includes/wikia/parser/templatetypes/handlers/NavboxTemplate.class.php [hunk 3]
```diff
@@ -28,24 +28,39 @@ private static function process( $html ) {
 		preg_match_all( $markerRegex, $html, $markers );
```
Comment (dianafa) [inline includes/wikia/parser/templatetypes/handlers/NavboxTemplate.class.php:28]: would it make sense to move this `$markerRegex` to const or a separate function to not to have to C&P it in tests?https://github.com/Wikia/app/pull/10466/files#diff-ed50d154df18b779157b998300265e66R28

**Comment by dianafa** [comment]
discussed offline, LGTM

**Pull request merged**